pennylane_lightning_kokkos/lightning_kokkos.py

```diff
@@ -276,6 +276,11 @@ def probability(self, wires=None, shot_range=None, bin_size=None):
         Returns:
             array[float]: list of the probabilities
         """
+        if wires and len(wires) > 1 and (not np.all(list(wires)[:-1] <= list(wires)[1:])):
```

Comment:
this check should really be done with `device_wires` below after line 291. Example of why:
```pycon
>>> dev = qml.device('lightning.qubit', wires=[2, 0, 1])
>>> obs = qml.PauliZ(2) @ qml.Identity(0) @ qml.Identity(1)
>>> dev.map_wires(obs.wires)  # should be accepted
<Wires = [0, 1, 2]>
>>> obs2 = qml.PauliZ(0) @ qml.Identity(1) @ qml.Identity(2)
>>> dev.map_wires(obs2.wires)
<Wires = [1, 2, 0]>  # should be rejected
```

Reply:
Yep, good call. It's probably worth adding the above test case to the pennylane `pl-device-tests` test runners, to ensure we catch such things early on external devices.

Reply:
Also, we will be be releasing with this forbidden behaviour now to allow time for a follow-up fix later in the 0.29 release cycle.